pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1265,6 +1265,32 @@ def compute_matrix() -> np.ndarray:  # pylint: disable=arguments-differ
         """
         return np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
 
+    @staticmethod
+    @lru_cache()
+    def compute_sparse_matrix() -> sparse.csr_matrix:  # pylint: disable=arguments-differ
+        r"""Sparse Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.SWAP.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.SWAP.compute_matrix())
+        <Compressed Sparse Row sparse matrix of dtype 'int64'
+                with 4 stored elements and shape (4, 4)>
+          Coords        Values
+          (0, 0)        1
+          (1, 2)        1
+          (2, 1)        1
+          (3, 3)        1
+        """
+        return sparse.csr_matrix([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
```

Comment:
```suggestion
        return sparse.csr_matrix(([1,1,1,1], [0,2,1,3], [0,1,2,3,4]))
```
Any advantage to using the dense representation over the `data,indices,indptr` rep?

Reply:
I agree that the latter option is a more concise representation. While I can see some situations where the dense representation would be easier to read, this isn't one of those cases since the matrix is presented in a single line.

Reply:
Yeah I took the dense repr only for readibility. They will end up the same

Reply:
Tried writing it multiline way, but black formatter won't allow lol. Let's take the latter one with a comment of the first expanded version